commit 41057ebde0025b0179b852dd785c9f3f0f08adad
Author: David Howells <dhowells@redhat.com>
Date:   Wed Sep 16 08:19:12 2020 +0100

    rxrpc: Support keys with multiple authentication tokens
    
    rxrpc-type keys can have multiple tokens attached for different security
    classes.  Currently, rxrpc always picks the first one, whether or not the
    security class it indicates is supported.
    
    Add preliminary support for choosing which security class will be used
    (this will need to be directed from a higher layer) and go through the
    tokens to find one that's supported.
    
    Signed-off-by: David Howells <dhowells@redhat.com>

diff --git a/net/rxrpc/conn_event.c b/net/rxrpc/conn_event.c
index aff184145ffa..bbf86203ed25 100644
--- a/net/rxrpc/conn_event.c
+++ b/net/rxrpc/conn_event.c
@@ -333,7 +333,8 @@ static int rxrpc_process_event(struct rxrpc_connection *conn,
 		if (ret < 0)
 			return ret;
 
-		ret = conn->security->init_connection_security(conn);
+		ret = conn->security->init_connection_security(
+			conn, conn->params.key->payload.data[0]);
 		if (ret < 0)
 			return ret;